core/domain/learner_progress_services.py

```diff
@@ -616,3 +648,118 @@ def get_collection_summary_dicts(collection_summaries):
         })
 
     return summary_dicts
+
+
+def get_activity_progress(user_id):
+    """Returns the progress of the learners - the explorations and collections
+    completed by the user and those in progress.
+
+    Args:
+        user_id: str. The id of the learner.
+
+    Returns:
```

Comment:
The more I think about it, the more strongly I lean towards a domain object. There is internal structure in the dict and the tuple-with-one-element-being-a-dict is starting to become a rather complicated "python blob". Some internal structure would be good.